src/components/data/data-utils.js

```diff
@@ -1397,10 +1397,10 @@ export const HeaderOptions = ({ sort, field, onSort, extraActions, children }) =
   ])
 
   return h(Sortable, {
-    sort, field, onSort
+    sort, field, onSort, columnMenu
```

Comment:
Not necessary to explore here, but I wonder if this could have also been solved by moving `columnMenu` outside of Sortable. Maybe by returning a Fragment.